src/panels/design/list-shmetro.tsx

```diff
@@ -27,30 +27,87 @@ const DesignListShmetro = () => {
+            paddingLeft: theme.spacing(5),
+        },
+    })
+);
+
+const StationNumberSHMetroLi = () => {
+    const { t } = useTranslation();
+    const classes = useStyles();
+    const dispatch = useAppDispatch();
+
+    const station_number = useAppSelector(store => store.param.showStationNumber);
+    const line_number = useAppSelector(store => store.param.line_num);
+
+    return useMemo(() => {
+        const handleSwitch = (_: React.ChangeEvent<HTMLInputElement>, checked: boolean) => {
+            if (checked) {
+                dispatch(setShowStationNumber(true));
+            } else {
+                dispatch(setShowStationNumber(false));
+            }
+        };
+
+        const handleChange = ({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+            dispatch(setLineNum(value));
+        };
+
+        return (
+            <>
+                <ListItem>
+                    <ListItemIcon>
+                        <Icon>account_balance_wallet</Icon>
+                    </ListItemIcon>
+                    <ListItemText primary={t('design.stationNumber')} />
+                    <ListItemSecondaryAction>
+                        <Switch color="primary" checked={station_number !== false} onChange={handleSwitch} />
+                    </ListItemSecondaryAction>
+                </ListItem>
+                <Collapse in={station_number !== false} unmountOnExit>
+                    <List component="div" disablePadding className={classes.nestedList}>
+                        <ListItem>
+                            <TextField placeholder={t('design.lineNum')} defaultValue={line_number} onChange={handleChange} />
+                        </ListItem>
+                    </List>
+                </Collapse>
+            </>
+        );
+    }, [line_number, station_number]);
+};
+
 const PanelTypeLi = () => {
     const { t } = useTranslation();
     const dispatch = useAppDispatch();
 
     const panelType = useAppSelector(store => store.param.info_panel_type);
-
     return useMemo(() => {
         const handleChange = ({ target: { value } }: ChangeEvent<{ name?: string; value: unknown }>) => {
             dispatch(setPanelType(value as PanelTypeShmetro));
         };
         return (
-            <ListItem>
-                <ListItemIcon>
-                    <Icon style={{ transform: 'rotate(180deg)' }}>credit_card</Icon>
-                </ListItemIcon>
-                <ListItemText primary={t('design.panelType.button')} />
-                <Select native value={panelType} onChange={handleChange} style={{ width: 166 }}>
-                    {Object.values(PanelTypeShmetro).map(type => (
-                        <option key={type} value={type}>
-                            {t('design.panelType.' + type)}
-                        </option>
-                    ))}
-                </Select>
-            </ListItem>
+            <>
+                <StationNumberSHMetroLi />
+                <ListItem>
```

Comment:
By having a `<StationNumberSHMetroLi />`, I would like to extract the following to a `<PanelTypeLi/>` like gzmtr. We can use a parent component to display different styles' info_panel.